apps/couch_httpd/src/couch_httpd.erl

```diff
@@ -121,16 +121,32 @@ start_link(Name, Options) ->
     % get the same value.
     couch_server:get_uuid(),
 
+    % add barrel log event handler
+    lager_handler_watcher:start(lager_event, barrel_log_event_handler, []),
+
+    WSLoop = fun(_Payload, [], _ReplyChannel) ->
+        ok
+    end,
+
     Loop = fun(Req)->
+        H = mochiweb_request:get_header_value("Upgrade", Req),
+        IsWebsocket = (H =/= undefined andalso string:to_lower(H) =:= "websocket"),
         case SocketOptions of
         [] ->
             ok;
         _ ->
             ok = mochiweb_socket:setopts(Req:get(socket), SocketOptions)
         end,
-        apply(?MODULE, handle_request, [
-            Req, DefaultFun, UrlHandlers, DbUrlHandlers, DesignUrlHandlers
-        ])
+
+        case IsWebsocket of
+        false -> apply(?MODULE, handle_request, [
+                    Req, DefaultFun, UrlHandlers, DbUrlHandlers, DesignUrlHandlers
+                 ]);
+        true -> {ReentryWs, ReplyChannel} = mochiweb_websocket:upgrade_connection(
+                                  Req, WSLoop),
+                gproc:reg({p, l, {log_event_handler, log}}, ReplyChannel),
```

Comment:
let's plane for the future. We can do more over websockets.

Instead of automatically subscribing, we should rather handle the subscription in the WS loop. Not sure yet what would be the final protocol. But we can start for now to answer on a specific command "SUBSCRIBE_LOGS" .


Thoughts?